src/main/java/de/tum/in/www1/exerciseapp/service/ExerciseService.java

```diff
@@ -118,4 +137,26 @@ public void delete(Long id, boolean deleteParticipations) {
         exerciseRepository.delete(id);
     }
 
+    /**
+     * Delete build plans (except BASE) of all exercise participations.
+     *
+     * @param id id of the exercise for which build plans in respective participations are deleted
+     */
+    @Transactional
+    public void deleteBuildPlans(Long id) {
+        log.debug("Request to delete build plans for Exercise : {}", id);
+        Exercise exercise = findOneLoadParticipations(id);
```

Comment:
Please check that this exercise is a ProgrammingExercise, otherwise log it do nothing

Reply:
Done.